### ISSUE TITLE
[data grid] Detail panel toggle header has conflicting aria-label and role="presentation"

### ISSUE THREAD

**Issue by jonathanraxa** [mui/mui-x#22140]
### Steps to reproduce

1. Navigate to [Data Grid - Master-detail row panels](https://mui.com/x/react-data-grid/master-detail/#customizing-the-detail-panel-toggle)
2. Run an axe accessibility scan (I used [axe DevTools](https://chromewebstore.google.com/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
3. Observe the violation on the detail panel toggle column header, we are specifically looking at: "Elements must only use permitted ARIA attributes"

The violation looks like this (see screenshot below) directed at this line: `<div aria-label="Detail panel toggle" role="presentation"></div>`

<img width="815" height="523" alt="Image" src="https://github.com/user-attachments/assets/42642892-c572-475a-bab0-589bd3f9e45e" />

### Current behavior 

`<div aria-label="Detail panel toggle" role="presentation"></div>` fails accessibility scan due to [Elements must only use permitted ARIA attributes](https://dequeuniversity.com/rules/axe/4.11/aria-prohibited-attr?application=AxeChrome)

<img width="993" height="556" alt="Image" src="https://github.com/user-attachments/assets/f2b7e3b7-2f27-4380-9550-50affc3ad6a7" />

### Expected behavior

No accessibility failure for "Elements must only use permitted ARIA attributes" to occur on the DataGrid

### Context

Something we stumbled across when running accessibility audits to our app and realized it was coming from [Data Grid - Master-detail row panels](https://mui.com/x/react-data-grid/master-detail/#customizing-the-detail-panel-toggle).

It seems like [PR #21634](https://github.com/mui/mui-x/pull/21634) added `role="presentation" ` to the detail panel toggle header content, but the existing `aria-label` was not removed.

Per the [WAI-ARIA spec (conflict resolution)](https://www.w3.org/TR/wai-aria-1.2/#conflict_resolution_presentation_none), when an element has global ARIA states or properties (like `aria-label`), user agents MUST ignore the `role="presentation"` and expose the element with its implicit role. This means `role="presentation"` has no effect here given that it's overridden by aria-label.

This causes an axe presentation-role-conflict violation, and browser behavior is inconsistent (see [w3c/aria#1824](https://github.com/w3c/aria/issues/1824)).

<details>

<summary>npx @mui/envinfo</summary>

- MUI Material: 7.3.10
- MUI X Data Grid Pro: 8.28.2
- MUI X Date Pickers: 8.27.2
- React: 19.2.5
- TypeScript: 5.9.3
- Node: 24.11.0

</details>

Thank you!

**Comment by michelengelen** [comment]
Hey @jonathanraxa ... I have opened a PR to address this issue! Thanks for raising it! 🙇🏼

**Comment by jonathanraxa** [comment]
Thank you @michelengelen !
